From 322c979f270df07d9a72297017081f1fee739def Mon Sep 17 00:00:00 2001
From: Benoit Parrot <bparrot@ti.com>
Date: Tue, 18 Jun 2013 17:18:31 -0500
Subject: [PATCH 1/4] gpu:drm:tilcdc: get preferred_bpp value from DT

The preferred_bpp value in currently hard-coded to 16.
This causes color corruption on the am335x-evm lcd panel which
requires 32 bpp instead. This changes attempts to use the configured
bpp value from the DT or built-in panel-info struct.

Signed-off-by: Benoit Parrot <bparrot@ti.com>
Acked-by: Rob Clark <robdclark@gmail.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
---
 drivers/gpu/drm/tilcdc/tilcdc_slave.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-3.12.24-rt38-r7s8/drivers/gpu/drm/tilcdc/tilcdc_slave.c
===================================================================
--- linux-3.12.24-rt38-r7s8.orig/drivers/gpu/drm/tilcdc/tilcdc_slave.c
+++ linux-3.12.24-rt38-r7s8/drivers/gpu/drm/tilcdc/tilcdc_slave.c
@@ -338,6 +338,8 @@ static int slave_probe(struct platform_d
 		return ret;
 	}
 
+	mod->preferred_bpp = slave_info.bpp;
+
 	i2c_node = of_find_node_by_phandle(i2c_phandle);
 	if (!i2c_node) {
 		dev_err(&pdev->dev, "could not get i2c bus node\n");
